File: 830#Positions_of_Large_Groups.py
class Solution:
    def largeGroupPositions(self, S):
        """
        :type S: str
        :rtype: List[List[int]]
        """
        if len(S) < 3:
            return []
        res = []
        i = 0
        while i < len(S) - 1:
            count = 1
            j = i + 1
            while j <= len(S) - 1 and S[j] == S[i]:
                j += 1
                count += 1
            if count >= 3:
                res.append([i, j-1])
            i = j
            count = 0
        return res

File: 830#Positions_of_Large_Groups.go
// 830. Positions of Large Groups

package main

func largeGroupPositions(S string) [][]int {
	if len(S) < 3 {
		return make([][]int, 0)
	}
	res := [][]int{}
	i := 0
	for i < len(S)-1 {
		count := 1
		j := i + 1
		for (j <= len(S)-1) && (S[j] == S[i]) {
			j++
			count++
		}
		if count >= 3 {
			pair := []int{i, j - 1}
			res = append(res, pair)
		}
		i = j
	}
	return res
}
